backend/apps/slack/commands/gsoc.py

```diff
@@ -5,9 +5,22 @@
 from apps.slack.apps import SlackConfig
 from apps.slack.blocks import markdown
 from apps.slack.commands.constants import COMMAND_START
-from apps.slack.constants import FEEDBACK_CHANNEL_MESSAGE, NL
+from apps.slack.constants import FEEDBACK_CHANNEL_MESSAGE
 
 COMMAND = "/gsoc"
+VALID_YEARS = range(2020, 2025)
+
+
+def get_gsoc_projects_by_year(year):
+    """Get GSOC projects for a specific year using search."""
+    from apps.owasp.api.search.project import get_projects
+
+    query = f"gsoc{year}"
+    return get_projects(
+        query=query,
+        attributes=["idx_name", "idx_url"],
+        limit=100,
+    )
```

Comment:
As I mentioned in the issue description we can use local DB data for getting the list. We don't need to query Algolia.

Move this logic to `Project::get_gsoc_projects(year)`